src/GenAPI.Desktop/Microsoft.DotNet.BuildTools.GenAPI.targets

```diff
@@ -2,9 +2,12 @@
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <GenAPITargetPath Condition="'$(GenAPITargetPath)' == ''">$(TargetDir)$(TargetName).cs</GenAPITargetPath>
-    <_GenAPIPath>$(MSBuildThisFileDirectory)\..\tools\GenApi.exe</_GenAPIPath>
+    <_GenAPIPath>$(MSBuildThisFileDirectory)\..\tools\GenAPI.exe</_GenAPIPath>
 
-    <!-- Sequence ourselves last in PrepareForRun 
+    <_GenAPICommand Condition=" '$(OS)' == 'Windows_NT'">"$(_GenAPIPath)"</_GenAPICommand>
+    <_GenAPICommand Condition=" '$(OS)' != 'Windows_NT'">mono --runtime=v4.0.30319 "$(_GenAPIPath)"</_GenAPICommand>
```

Comment:
I'm fine introducing the extension point and the casing but I don't really like making the assumption that not windows means mono. In the majority of our tools not windows implies .NET Core and not mono. With the extension point you should be able to set _GenAPICommand in your project or common targets to get this setup for mono.